test/python/transpiler/test_elide_swaps.py

```diff
@@ -0,0 +1,128 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test ElideSwap pass"""
+
+import unittest
+
+from qiskit import QuantumCircuit
+from qiskit.transpiler.passes import ElideSwaps
+from qiskit.test import QiskitTestCase
+
+
+class TestElideSwaps(QiskitTestCase):
+    """Test swap elision logical optimization pass."""
```

Comment:
These look generally solid for the main cases.  Maybe an extra two tests:

- swaps inside control-flow blocks are ignored
- `swap().c_if(...)` is ignored.

Reply:
Added in: https://github.com/Qiskit/qiskit-terra/pull/9523/commits/8dfd4f7b7e14b056f1ba29e232ac4c212024ceef